docs/_docs/04_reference/config-blocks-and-attributes.md

```diff
@@ -393,6 +394,63 @@ inputs = {
 }
 ```
 
+**What is deep merge?**
```

Comment:
NIT: consider adding an example showing two `terragrunt.hcl` files and what a merged output would look like.

Reply:
Added example: https://github.com/gruntwork-io/terragrunt/pull/1764